runtime/parser/statement_test.go

```diff
@@ -2684,6 +2684,11 @@ func TestSoftKeywordsInStatement(t *testing.T) {
 	}
 
 	for _, keyword := range softKeywords {
+		// it's not worth the additional complexity to support assigning to `remove` or `attach`-named
+		// variables, so we just accept this as a parsing error
+		if keyword == keywordAttach || keyword == keywordRemove {
+			continue
+		}
```

Comment:
Maybe replace this with a different test above: Instead of checking the soft keyword in a statement, check it e.g. in a function declaration. That should never conflict, even for attach and remove statements

Reply:
I want to leave this here just to document the behavior, but I will add an additional test for function decls as well.